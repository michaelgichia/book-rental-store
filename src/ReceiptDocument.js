/**
 * Document to print
 */
import React from "react";
import { Table, Row, Col, Divider } from "antd";
import { handlePriceCalculation, handleTotal } from "./RentedBooks";
export function ReceiptDocument({ dataSource, user }) {
  return (
    <Row className="printme">
      <Col span={24}>
        <Divider orientation="left">Personal information</Divider>
      </Col>
      <Col span={24}>
        <p>
          {user.firstName || ""} {user.lastName || ""}
        </p>
      </Col>
      <Col span={24}>
        <p>{user.phoneNumber || ""}</p>
      </Col>
      <Col span={24}>
        <p>{user.email || ""}</p>
      </Col>
      <Col span={24}>
        <Divider orientation="left">Books</Divider>
      </Col>
      <Table
        columns={[
          {
            title: "Title",
            dataIndex: "title",
            ellipsis: true,
            width: "50%",
          },
          {
            title: "Return date",
            key: "returnDate",
            dataIndex: "returnDate",
            align: "right",
          },
          {
            title: "Amount",
            key: "amount",
            dataIndex: "amount",
            align: "right",
            render: (_, book) => handlePriceCalculation(book),
          },
        ]}
        dataSource={dataSource}
        pagination={false}
        rowKey="id"
        size="small"
        footer={() => (
          <div className="print-doc-footer">
            <h3>Total: $ {handleTotal(dataSource)}</h3>
          </div>
        )}
      />
    </Row>
  );
}

export default ReceiptDocument;
