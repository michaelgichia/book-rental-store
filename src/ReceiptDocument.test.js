import React from "react";
import { shallow, mount } from "enzyme";

import ReceiptDocument from "./ReceiptDocument";

describe("<ReceiptDocument />", () => {
  const rentedBooks = [
    {
      amount: 1,
      author: "Arielle Howe",
      id: "KXWxtmgYNC",
      imageUrl: "http://placeimg.com/220/280/tech",
      returnDate: "2020-09-15",
      title: "Shocking for hurl precipitate",
    },
  ];
  const user = {
    firstName: "Michael",
    lastName: "Gichia",
    phoneNumber: "0710853398",
    email: "gichuru.gichia@gmail.com",
  };

  it("should test ReceiptDocument component with default state of empty array", () => {
    const component = shallow(
      <ReceiptDocument dataSource={[]} user={user} display={true} />
    );
    expect(component).toMatchSnapshot();
  });

  it("should test ReceiptDocument component without user information", () => {
    const component = shallow(
      <ReceiptDocument dataSource={[]} user={{}} display={true} />
    );
    expect(component).toMatchSnapshot();
  });

  it("should display rented books", () => {
    shallow(
      <ReceiptDocument dataSource={rentedBooks} user={user} display={true} />
    );
  });
});
