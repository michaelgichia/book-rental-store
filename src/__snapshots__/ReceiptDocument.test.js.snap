// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReceiptDocument /> should test ReceiptDocument component with default state of empty array 1`] = `
<Row
  className="printme"
>
  <Col
    span={24}
  >
    <Divider
      orientation="left"
    >
      Personal information
    </Divider>
  </Col>
  <Col
    span={24}
  >
    <p>
      Michael
       
      Gichia
    </p>
  </Col>
  <Col
    span={24}
  >
    <p>
      0710853398
    </p>
  </Col>
  <Col
    span={24}
  >
    <p>
      gichuru.gichia@gmail.com
    </p>
  </Col>
  <Col
    span={24}
  >
    <Divider
      orientation="left"
    >
      Books
    </Divider>
  </Col>
  <Table
    columns={
      Array [
        Object {
          "dataIndex": "title",
          "ellipsis": true,
          "title": "Title",
          "width": "50%",
        },
        Object {
          "align": "right",
          "dataIndex": "returnDate",
          "key": "returnDate",
          "title": "Return date",
        },
        Object {
          "align": "right",
          "dataIndex": "amount",
          "key": "amount",
          "render": [Function],
          "title": "Amount",
        },
      ]
    }
    dataSource={Array []}
    footer={[Function]}
    pagination={false}
    rowKey="id"
    size="small"
  />
</Row>
`;

exports[`<ReceiptDocument /> should test ReceiptDocument component without user information 1`] = `
<Row
  className="printme"
>
  <Col
    span={24}
  >
    <Divider
      orientation="left"
    >
      Personal information
    </Divider>
  </Col>
  <Col
    span={24}
  >
    <p>
       
    </p>
  </Col>
  <Col
    span={24}
  >
    <p />
  </Col>
  <Col
    span={24}
  >
    <p />
  </Col>
  <Col
    span={24}
  >
    <Divider
      orientation="left"
    >
      Books
    </Divider>
  </Col>
  <Table
    columns={
      Array [
        Object {
          "dataIndex": "title",
          "ellipsis": true,
          "title": "Title",
          "width": "50%",
        },
        Object {
          "align": "right",
          "dataIndex": "returnDate",
          "key": "returnDate",
          "title": "Return date",
        },
        Object {
          "align": "right",
          "dataIndex": "amount",
          "key": "amount",
          "render": [Function],
          "title": "Amount",
        },
      ]
    }
    dataSource={Array []}
    footer={[Function]}
    pagination={false}
    rowKey="id"
    size="small"
  />
</Row>
`;
